lib/teiserver_web/controllers/o_auth/code_controller.ex

```diff
@@ -0,0 +1,96 @@
+defmodule TeiserverWeb.OAuth.CodeController do
+  use TeiserverWeb, :controller
+  alias Teiserver.OAuth
+
+  # https://www.rfc-editor.org/rfc/rfc6749.html#section-4.1.3
+  @spec token(Plug.Conn.t(), %{}) :: Plug.Conn.t()
+
+  def token(conn, %{"grant_type" => "authorization_code"} = params) do
+    case Enum.find(
+           ["client_id", "code", "redirect_uri", "client_id", "code_verifier", "grant_type"],
+           fn key -> not Map.has_key?(params, key) end
+         ) do
+      nil ->
+        exchange_token(conn, params)
+
+      missing_key ->
+        conn |> put_status(400) |> render(:error, error_description: "missing #{missing_key}")
```

Comment:
Doesn't like like error is generated according to https://www.rfc-editor.org/rfc/rfc6749.html#section-4.1.4 here and below ?

Reply:
I've amended it to return `{"error": "invalid_request"}` [at least](https://github.com/geekingfrog/teiserver/blob/3514c247d065f9b028c50923e390ecc919d9513f/test/teiserver_web/controllers/o_auth/code_controller_test.exs#L104).